### PR TITLE
lexer raise for literals followed by alphanum

### PR DESCRIPTION
**Pull request by hhugo**
http://caml.inria.fr/mantis/view.php?id=6816

### CODE REVIEW

File: parsing/lexer.mli [hunk 1]
```diff
@@ -24,6 +24,7 @@ type error =
   | Unterminated_string_in_comment of Location.t * Location.t
   | Keyword_as_label of string
   | Literal_overflow of string
+  | Literal_malformed of string
```
Comment (gasche) [inline parsing/lexer.mli]: `Literal_malformed` does not sound like proper english to me. It should be `Malformed_literal` -- or maybe the more neutral `Invalid_literal`.

Comment (hhugo) [inline parsing/lexer.mli]: True. This was only to get a common prefix with `Literal_overflow`.
`Invalid_literal` seems better anyway.

File: parsing/lexer.mll [hunk 2]
```diff
@@ -349,6 +352,8 @@ rule token = parse
           NATIVEINT (cvt_nativeint_literal (Lexing.lexeme lexbuf))
         with Failure _ ->
           raise (Error(Literal_overflow "nativeint", Location.curr lexbuf)) }
+  | (float_literal | int_literal) identchar*
```
Comment (planar) [inline parsing/lexer.mll]: What's the point of the star here? You want to raise as soon as you see one `identchar`, and you don't want to raise if there are none.

Comment (hhugo) [inline parsing/lexer.mll]: The star should probably be a plus.
The error `Invalid literal 0o18` seems weird when trying to lex `0o1823`

Comment (planar) [inline parsing/lexer.mll]: Indeed, a plus would be more robust.

**Comment by gasche** [comment]
Merged in trunk. Thanks!

**Comment by damiendoligez** [comment]
FTR, this change broke (at least) the following OPAM packages:
- archimedes 0.4.17
- caml2html 1.4.3
- cubicle 1.0.2

**Comment by gasche** [comment]
Would you have the code snippets that break? Just curious.

**Comment by damiendoligez** [comment]
For archimedes:

```
               let cosa = cos angle /.2.
               and sina = sin angle  /.2.in
```

For cubicle:

```
  let max_id_vars = !i - 1in
```

For caml2html, it's in a file designed to torture-test the parser, see https://github.com/mjambon/caml2html/pull/3

**Pull request closed without merging**